### PR TITLE
[PSR-12] Covered typed properties

### PR DESCRIPTION
**Pull request by samdark**

### CODE REVIEW

File: proposed/extended-coding-style-guide.md [hunk 1]
```diff
@@ -396,6 +396,8 @@ Property names MUST NOT be prefixed with a single underscore to indicate
 protected or private visibility. That is, an underscore prefix explicitly has
 no meaning.
 
+When present, type declaration MUST precede property name.
```
Comment (JanTvrdik) [inline proposed/extended-coding-style-guide.md]: I don't understand, where else can the type declaration be?

Comment (samdark) [reply proposed/extended-coding-style-guide.md]: `public int static $test`?

Comment (Jean85) [reply proposed/extended-coding-style-guide.md]: Then you must be more precise:
```suggestion
When present, type declaration MUST immediately precede property name.
```

Comment (samdark) [reply proposed/extended-coding-style-guide.md]: Applied change. Actually I'm not sure such order is allowed by PHP. Had no time to check source code or get PHP 7.4 working locally. Anyone willing to do so?

Comment (JanTvrdik) [reply proposed/extended-coding-style-guide.md]:
```
docker run --rm -it devilbox/php-fpm-7.4:latest php -r 'class Test { public int static $bar; }
```

results in

```
Parse error: syntax error, unexpected 'static' (T_STATIC), expecting variable (T_VARIABLE) in Command line code on line 1
```

Comment (samdark) [reply proposed/extended-coding-style-guide.md]: Thank you! Then the rule isn't needed at all.

**Comment by samdark** [comment]
@JanTvrdik do you mean the following is allowed?

```php
class X
{
    public int$test;
```
?

**Comment by Jean85** [comment]
Yes it is:
```
$ docker run --rm devilbox/php-fpm-7.4 php -r 'class Test { public static int$bar; }; echo "OK";'
OK%                                                                                                                                                                                                        
```

**Comment by samdark** [comment]
Adjusted.

**Pull request merged**